### ISSUE TITLE
Fix Dapper MySQLTests

### ISSUE THREAD

**Issue by bgrainger** [mysql-net/MySqlConnector#261]
Dapper includes a [MySQLTests](https://github.com/StackExchange/Dapper/blob/master/Dapper.Tests/Providers/MySQLTests.cs) test suite.

If I change the tests to use the `MySqlConnector` NuGet package instead of `MySql.Data`, `DapperEnumValue_Mysql` fails with:

```
System.NotImplementedException : GetFieldType for Null is not implemented
   at MySql.Data.MySqlClient.Results.ResultSet.GetFieldType(Int32 ordinal)
   at MySql.Data.MySqlClient.MySqlDataReader.GetFieldType(Int32 ordinal)
   at Dapper.SqlMapper.GetColumnHash(IDataReader reader, Int32 startBound, Int32 length)
   at Dapper.SqlMapper.QueryRowImpl[T](IDbConnection cnn, Row row, CommandDefinition& command, Type effectiveType)
   at Dapper.SqlMapper.QuerySingle[T](IDbConnection cnn, String sql, Object param, IDbTransaction transaction, Nullable`1 commandTimeout, Nullable`1 commandType)
   at Dapper.Tests.Common.DapperEnumValue(IDbConnection connection)
   at Dapper.Tests.MySQLTests.DapperEnumValue_Mysql()
```

**Comment by bgrainger** [comment]
The failing line of code is

```
(AnEnum)connection.QuerySingle<int>("select @v, @y, @z", new { v = AnEnum.B, y = (AnEnum?)AnEnum.B, z = (AnEnum?)null });
```

Connector/NET returns `VARCHAR` from `GetDataTypeName(2)` and `typeof(string)` from `GetFieldType(2)`. MySqlConnector throws a `NotImplementedException` in both functions.

**Comment by bgrainger** [comment]
Fixed in [0.19.4](https://www.nuget.org/packages/MySqlConnector/0.19.4)
